Xamarin.PropertyEditing.Mac/Controls/RectangleEditorControl.cs

```diff
@@ -0,0 +1,57 @@
+﻿using System;
+using System.Collections;
+using System.Diagnostics;
+using System.Drawing;
+using System.Reflection;
+using AppKit;
+using CoreGraphics;
+using Xamarin.PropertyEditing.ViewModels;
+
+namespace Xamarin.PropertyEditing.Mac
+{
+	internal class RectangleEditorControl<T> : BaseRectangleEditorControl<T>
+		where T : struct
+	{
+		public RectangleEditorControl ()
+		{
+			XLabel.Frame = new CGRect (0, -5, 25, 20);
+			XLabel.Value = "X:";
+
+			XEditor.Frame = new CGRect (25, 0, 50, 20);
+
+			YLabel.Frame = new CGRect (85, -5, 25, 20);
+			YLabel.Value = "Y:";
+
+			YEditor.Frame = new CGRect (105, 0, 50, 20);
+
+			WidthLabel.Frame = new CGRect (160, -5, 45, 20);
+			WidthLabel.Value = "Width:";
+
+			WidthEditor.Frame = new CGRect (205, 0, 50, 20);
+
+			HeightLabel.Frame = new CGRect (260, -5, 45, 20);
+			HeightLabel.Value = "Height:";
+
+			HeightEditor.Frame = new CGRect (305, 0, 50, 20);
+		}
+
+		protected override void UpdateModelValue ()
```

Comment:
Alternatively and probably better, there aren't enough rectangle types to worry about, a type-based cache of delegates to the getters is probably enough.